interpolate/d2v-interpolate-common/src/main/kotlin/io/data2viz/interpolate/number.kt

```diff
@@ -1,34 +1,32 @@
 package io.data2viz.interpolate
 
+import io.data2viz.math.*
 import kotlin.math.round
 
 /**
- * An interpolator transforms a normalized continuous range (0.0 -> 1.0)
- * to a Domain object R
+ * An interpolator transforms a normalized continuous range (0% -> 100%) to an object T
  */
-typealias Interpolator<R> = (Double) -> R
+typealias Interpolator<T> = (Percent) -> T
 
-// TODO : remove use only double
-fun interpolateNumber(a: Number, b: Number): Interpolator<Double> {
-    val diff = b.toDouble() - a.toDouble()
-    return { t -> a.toDouble() + t * diff }
-}
+/**
+ * An un-interpolator transforms an object T to a normalized continuous range (0% -> 100%)
+ */
+typealias UnInterpolator<T> = (T) -> Percent
 
-fun interpolateNumber(a: Double, b: Double): Interpolator<Double>{
-    val diff = b - a
-    return { t -> a + t * diff }
+fun interpolateNumber(start: Double, end: Double): Interpolator<Double>{
+    val diff = end - start
+    return { percent -> start + percent.value * diff }
```

Comment:
There is no need of this intermediate val. It can then be simplified:
```
fun interpolateNumber(start: Double, end: Double): Interpolator<Double> =
    { percent -> start + percent * (end - start) }
```